src/emulator/system.c

```diff
@@ -0,0 +1,1801 @@
+        N64_BTN_Z,      // GCN_BTN_L
+        N64_BTN_R,      // GCN_BTN_R
+        N64_BTN_CRIGHT, // GCN_BTN_Z
+        N64_BTN_START,  // GCN_BTN_START
+        0x08000000,     // GCN_BTN_UNK8
+        0x04000000,     // GCN_BTN_UNK9
+        0x02000000,     // GCN_BTN_UNK10
+        0x01000000,     // GCN_BTN_UNK11
+        N64_BTN_CUP,    // GCN_BTN_DPAD_UP
+        N64_BTN_CUP,    // GCN_BTN_DPAD_DOWN
+        N64_BTN_CUP,    // GCN_BTN_DPAD_LEFT
+        N64_BTN_CUP,    // GCN_BTN_DPAD_RIGHT
+        N64_BTN_CUP,    // GCN_BTN_CSTICK_UP
+        N64_BTN_CDOWN,  // GCN_BTN_CSTICK_DOWN
+        N64_BTN_CLEFT,  // GCN_BTN_CSTICK_LEFT
+        N64_BTN_CRIGHT, // GCN_BTN_CSTICK_RIGHT
+    },
+};
+// clang-format on
+
+static SystemDevice gaSystemDevice[] = {
+    {
+        SOT_HELP,
+        &gClassHelpMenu,
+        0,
+        {0},
+    },
+    {
+        SOT_FRAME,
+        &gClassFrame,
+        0,
+        {0},
+    },
+    {
+        SOT_LIBRARY,
+        &gClassLibrary,
+        0,
+        {0},
+    },
+    {
+        SOT_CODE,
+        &gClassCode,
+        0,
+        {0},
+    },
+    {
+        SOT_AUDIO,
+        &gClassAudio,
+        0,
+        {0},
+    },
+    {
+        SOT_VIDEO,
+        &gClassVideo,
+        0,
+        {0},
+    },
+    {
+        SOT_CONTROLLER,
+        &gClassController,
+        0,
+        {0},
+    },
+    {
+        SOT_CPU,
+        &gClassCPU,
+        1,
+        {{0, 0x00000000, 0xFFFFFFFF}},
+    },
+    {
+        SOT_RAM,
+        &gClassRAM,
+        3,
+        {{256, 0x00000000, 0x03EFFFFF}, {2, 0x03F00000, 0x03FFFFFF}, {1, 0x04700000, 0x047FFFFF}},
+    },
+    {
+        SOT_RDP,
+        &gClassRDP,
+        2,
+        {{0, 0x04100000, 0x041FFFFF}, {1, 0x04200000, 0x042FFFFF}},
+    },
+    {
+        SOT_RSP,
+        &gClassRSP,
+        1,
+        {{0, 0x04000000, 0x040FFFFF}},
+    },
+    {
+        SOT_MI,
+        &gClassMI,
+        1,
+        {{0, 0x04300000, 0x043FFFFF}},
+    },
+    {
+        SOT_VI,
+        &gClassVI,
+        1,
+        {{0, 0x04400000, 0x044FFFFF}},
+    },
+    {
+        SOT_AI,
+        &gClassAI,
+        1,
+        {{0, 0x04500000, 0x045FFFFF}},
+    },
+    {
+        SOT_PI,
+        &gClassPI,
+        1,
+        {{0, 0x04600000, 0x046FFFFF}},
+    },
+    {
+        SOT_SI,
+        &gClassSI,
+        1,
+        {{0, 0x04800000, 0x048FFFFF}},
+    },
+    {
+        SOT_RDB,
+        &gClassRdb,
+        1,
+        {{0, 0x04900000, 0x0490FFFF}},
+    },
+    {
+        SOT_DISK,
+        &gClassDisk,
+        2,
+        {{0, 0x05000000, 0x05FFFFFF}, {1, 0x06000000, 0x06FFFFFF}},
+    },
+    {
+        SOT_ROM,
+        &gClassROM,
+        2,
+        {{0, 0x10000000, 0x1FBFFFFF}, {1, 0x1FF00000, 0x1FF0FFFF}},
+    },
+    {
+        SOT_PIF,
+        &gClassPIF,
+        1,
+        {{0, 0x1FC00000, 0x1FC007FF}},
+    },
+    {
+        SOT_NONE,
+        NULL,
+        0,
+        {0},
+    },
+};
+
+#include "lbl_8016FEA0.inc"
```

Comment:
just curious, what is this? maybe add a short comment?

Reply:
it's hacks used by diddy kong racing, for some reasons it breaks systemSetupGameALL to use this symbol directly so for now I'm using the address and I block the reloc in the config

Reply:
nvm looks like it's also used by paper mario, I need to dig into this to figure out if both usage are real lol, I'm a bit confused by this rught now